package.json

```diff
@@ -16,7 +16,8 @@
   ],
   "scripts": {
     "coverage": "cat coverage/lcov.info | coveralls",
-    "format": "npm run lint:fix || true",
+    "format": "npm run lint:fix && npm run prettier || true",
+    "prettier": "prettier src test --write",
```

Comment:
Here's what I do in my newer projects: let ESLint take care of all JavaScript formatting (using the Prettier plugin), since it has more comprehensive options for fixing/formatting stuff. e.g. it will apply Prettier for you *plus* any other formatting/fixes it does.

Then run Prettier only on non-JavaScript files that ESLint doesn't take care of, like Markdown and CSS.

so this would be more like

```
"prettier": "prettier \"**/*.md\" --write"
```

Reply:
(the reason to prefer it this way integrating Prettier w/ ESLint is there's a single source of truth for formatting the code, and it'll then be one atomic change per file rather than two-pass formatting)

Reply:
Now I think I get where you're coming from.

@exogen When I try that, when I run `npm run format`, I get this:

```diff
diff --git a/src/loadSchemaJSON.js b/src/loadSchemaJSON.js
index 001d311..e52297a 100644
--- a/src/loadSchemaJSON.js
+++ b/src/loadSchemaJSON.js
@@ -9,10 +9,8 @@ const DEFAULT_GRAPHQL = graphql

 function readFile(filename) {
   return new Promise((resolve, reject) => {
-    fs.readFile(
-      filename,
-      'utf8',
-      (err, data) => (err ? reject(err) : resolve(data))
+    fs.readFile(filename, 'utf8', (err, data) =>
+      err ? reject(err) : resolve(data)
     )
   })
 }
```

How established in this convention in open source projects?  (of invoking prettier via eslint?)

Prettier by itself seems to format differently than eslint + prettier.  

What do you think it friendlier to projects? Generally I autoformat with prettier, but I don't do anything built-in with eslint due to its performance.

Reply:
Maybe you'll have me converted to using eslint w/ prettier baked inside. 

I've such an engrained habit of using vanilla prettier...  So now we're kind of bikeshedding and its my personal habit.

Also my editors automatically invoke prettier. Ugh. 

Contextual question: Are you running eslint fixes (incl. prettier formatting) on save in your editor?  I haven't tried eslint on individual files, and haven't used fixers.